os/board/rtl8721csm/src/component/common/mbed/targets/hal/rtl8721d/flash_api.c

```diff
@@ -384,6 +378,16 @@ int  flash_stream_write(flash_t *obj, u32 address, u32 len, u8 * data)
 		}
 
 		addr_begin = (((addr_begin-1) >> 2) + 1) << 2;
+		for(;size >= 256 ;size -= 256){
+			_memcpy(write_data, buffer, 256);

```

Comment:
I still can't understand why memcpy is needed here, can't we use data(3rd parameter) directly as I   commented above?

Reply:
Hi @Taejun-Kwon , thank you for the advice. I have tried your suggested change for this part using the while loop instead of for loop. It seems that this change does not significantly improve the writing speed. Therefore, we decided to keep this code as it was as much as possible.

Reply:
My question is why extra 256bytes, write_data & memcpy are necessary?

Reply:
Hi @Taejun-Kwon ,
Based on my opinion and some tests I did, there is no difference in using memcpy instead of directly using the data. But since you are keen to know, I have already forwarded the question to its original author to investigate the initial considerations and make sure it can be safely replaced. 
So far I have not received the response, I will update to you once I know the result, thanks!

Reply:
I have received the response that it can be safely replaced with your suggested way. I have updated this change in the newest commit, thanks!